lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -1900,42 +1944,67 @@ GLStatus TFGraphLowering::visitGraphOperationInst(GraphOperationInst *inst) {
         inst->dump();
         llvm_unreachable("dtype attr must have been processed!");
       }
+      auto dtype = (TF_DataType)dtypeAttr;
+      auto dtypeSize = TF_DataTypeSize(dtype);
+
       // Tensor can support two cases: an array case (not yet implemented), and
       // a scalar case.
       SmallVector<APInt, 4> elements;
-      SmallVector<int64_t, 4> shape;
 
-      // The scalar case is very simple, the shape of a scalar is 0d, and the
-      // data type comes from an attr that should already be processed.
-      auto addScalar = [&](SymbolicValue value) {
+      // Add a scalar to the elements list, checking that it is the right size
```

Comment:
"checking that it is the right size" -> should we update the comment as we are actually fixing the sizes in the impl?